static/app/utils/useSyncedLocalStorageState.tsx

```diff
@@ -0,0 +1,49 @@
+import {useCallback, useEffect} from 'react';
+
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+
+type SyncedLocalStorageEvent = CustomEvent<{key: string; value: any}>;
+
+const SET_VALUE_EVENT_NAME = 'synced-local-storage';
+
+/**
+ * Same as `useLocalStorageState`, but notifies and reacts to state changes.
+ * Use this you want to access local storage state from multiple components
+ * on the same page.
+ */
+export function useSyncedLocalStorageState<S>(
+  key: string,
+  initialState: S | ((value?: unknown, rawValue?: unknown) => S)
+): [S, (value: S) => void] {
+  const [value, setValue] = useLocalStorageState(key, initialState);
+
+  const setValueAndNotify = useCallback(
+    newValue => {
+      setValue(newValue);
+
+      // We use a custom event to notify all consumers of this hook
+      window.dispatchEvent(
+        new CustomEvent(SET_VALUE_EVENT_NAME, {detail: {key, value: newValue}})
+      );
+    },
+    [key, setValue]
+  );
+
+  useEffect(() => {
+    const handleNewSyncedLocalStorageEvent = (event: Event) => {
+      const eventDetail = (event as SyncedLocalStorageEvent).detail;
```

Comment:
Type guard maybe :)? Then below on L36 you could just eventDetail.key ===

Reply:
Added a type guard - agree that this is much nicer!